scheduler/util.go

```diff
@@ -599,6 +599,11 @@ func tasksUpdated(jobA, jobB *structs.Job, taskGroup string) bool {
 		} else if !ar.Devices.Equal(&br.Devices) {
 			return true
 		}
+
+		// Inspect Identity being exposed
+		if !reflect.DeepEqual(at.Identity, bt.Identity) {
+			return true
+		}
```

Comment:
nitpick: prefer to avoid `reflect.DeepEqual` which is both slow and dangerous - we've fixed at least half a dozen bugs where `DeepEqual` treats things like nil vs empty slice differently.

Reply:
Good call. Added a WorkloadIdentity.Equal method + a test, and used that method instead of DeepEqual here.